node_group_advanced.tf

```
@@ -90,7 +90,37 @@ resource "aws_launch_template" "quortex_launch_tpl" {
 
   update_default_version = true
 
-  user_data = base64encode(
+  user_data = each.value.warm_pool_enabled ? base64encode(
+    templatefile(
+      "${path.module}/userdata-warm.sh.tpl",
+      {
+        cluster_name       = aws_eks_cluster.quortex.name
+        base64_cluster_ca  = aws_eks_cluster.quortex.certificate_authority[0].data
+        api_server_url     = aws_eks_cluster.quortex.endpoint
+        kubelet_extra_args = lookup(each.value, "kubelet_extra_args", "")
+        // define the k8s node taints (passed to --kubelet-extra-args)
+        node_taints = length(each.value.taints) == 0 ? "" : join(",", [for k, v in lookup(each.value, "taints", {}) : "${k}=${v}"])
+        // define the k8s node labels (passed to --kubelet-extra-args)
+        node_labels = join(
+          ",",
+          [
+            for k, v in
+            merge(
+              # Built-in labels
+              {
+                "eks.amazonaws.com/nodegroup-image" = local.ami_id_worker,
+                "eks.amazonaws.com/nodegroup"       = each.key,
+                "nodegroup"                         = each.key
+              },
+              # User-specified labels
+              lookup(each.value, "labels", {}),
+            )
+          : "${k}=${v}"]
+        )
+        use_max_pods = var.node_use_max_pods
+      }
+    )
+  ) : base64encode(
     templatefile(
       "${path.module}/userdata.sh.tpl",
```

Something like this would be easier to read

```suggestion
  user_data = base64encode(
    templatefile(
      each.value.warm_pool_enabled ? "${path.module}/userdata-warm.sh.tpl" : "${path.module}/userdata.sh.tpl",
```

---

Using a single template file would even be preferable as it would prevent oversights in future edits to these files.